### ISSUE TITLE
softrelu activation clipping bug in MKLDNN

### ISSUE THREAD

**Issue by safrooze** [apache/incubator-mxnet#11804]
## Description
With MKLDNN (enabled in mxnet-cuxxmkl builds), using softrelu activation, the output is clipped to maximum value of 88.37625885009766. This regression was introduced in 1.2.0 when MKLDNN was enabled for activations other than relu in [#10336](https://github.com/apache/incubator-mxnet/pull/10336). 

Looking at MKLDNN [implementation of softrelu](https://github.com/intel/mkl-dnn/blob/72236df2da36d2c73ce1c719c6e9fa04a2f6bdc7/src/common/math_utils.hpp#L168), I don't see any overflow prevention logic, so most likely the exp() function overflows and results in this bug.

## Environment info (Required)
mxnet-cu90mkl >= 1.2.0b20180403

## Minimum reproducible example
```
nd.Activation(nd.array([100]), act_type='softrelu')
```
output:
```
[ 88.37625885]
<NDArray 1 @cpu(0)>
```

@jinhuang415

**Comment by safrooze** [comment]
@pengzhao-intel @zheng-da

**Comment by ZhennanQin** [comment]
@safrooze Thanks for reporting this. We confirmed this is a bug in mkldnn, and will fix this in future. considering input > 88 is a corner case in most of models, most users shouldn't face this problem when using soft_relu with mkldnn. If you're having trouble with this on a certain model, please let us know.

**Comment by azai91** [comment]
@ZhennanQin do we have an ETA when this would be fixed? should we fallback if input too large?

**Comment by ZhennanQin** [comment]
@azai91 , this defect will be addressed within the next sprint, with expected target date ww33.5.

**Comment by safrooze** [comment]
@ZhennanQin Is this fixed in MKLDNN now?

**Comment by ZhennanQin** [comment]
@safrooze  Patch is ready for internal test. Will push to github repo in next promotion.

**Comment by apeforest** [comment]
@safrooze Please verify at your earliest convenience if this issue has been fixed. Thanks!

**Comment by apeforest** [comment]
@safrooze I have verified that using the master from intel/mkl-dnn module the problem has been fixed.
`>>> from mxnet import ndarray as nd
>>> nd.Activation(nd.array([100]), act_type='softrelu')

[100.]
<NDArray 1 @cpu(0)>`
@sandeep-krishnamurthy @nswamy Please close this issue.

**Comment by safrooze** [comment]
@apeforest Thanks! Closing this issue. Just for reference this fix will be included in MKLDNN 0.17 release.
